webpack-subresource-integrity-embroider/index.js

```diff
@@ -23,7 +23,20 @@ class SubresourceIntegrityPlugin {
         );
         const scriptElements =
           indexHtml.window.document.querySelectorAll("script");
-        const linkElements = indexHtml.window.document.querySelectorAll("link");
+        const linkElements = Array.from(
+          indexHtml.window.document.querySelectorAll("link")
+        ).filter(element => {
+            const relationship = element.getAttribute("rel");
+            // link integrity is only valid for rel="stylesheet", "module", "modulepreload"
+            if (
+              relationship === "stylesheet"
+              || relationship === "module"
+              || relationship === "modulepreload"
+            ) {
```

Comment:
I think there is a big here. `rel` attribute may be a space-separated list of relationships. Not only a single value.

I only found one legitim combination of `stylesheet`, `module`, and `modulepreload`: [Alternate style sheets](https://developer.mozilla.org/en-US/docs/Web/CSS/Alternative_style_sheets).
```html
<link href="fancy.css" rel="alternate stylesheet" title="Fancy" />
```

I don't think alternative style sheets are really used in production. Especially if looking at browser support. But the code is likely more stable if we encounter for that valid scenario.

Additionally we could move the filter into CSS selector for readability:

```js
indexHtml.window.document.querySelectorAll(
  'link:is([rel=~"module"], [rel=~"module preload"], [rel=~"stylesheet"])'
)
```

Reply:
TIL about alternate style sheets, I'll adjust this - thanks!

Reply:
Updated in https://github.com/jelhan/webpack-subresource-integrity-embroider/pull/16/commits/f2843c464cf898f3b11df6326ce59878405fbb7b !